# Amazon Sales Analysis

### Importing Required Libraries for 

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt 
%matplotlib inline
import seaborn as sns

### Loading Dataset of Amazon Sales

In [3]:
df=pd.read_csv('Amazon_Sales_Data.csv',encoding= 'unicode_escape')

### TOP 5 Data in dataset

In [4]:
df.head()

index             Order ID      Date                        Status  \
0      0  405-8078784-5731545  04-30-22                     Cancelled   
1      1  171-9198151-1101146  04-30-22  Shipped - Delivered to Buyer   
2      2  404-0687676-7273146  04-30-22                       Shipped   
3      3  403-9615377-8133951  04-30-22                     Cancelled   
4      4  407-1069790-7240320  04-30-22                       Shipped   

  Fulfilment Sales Channel ship-service-level  Category Size Courier Status  \
0   Merchant     Amazon.in           Standard   T-shirt    S     On the Way   
1   Merchant     Amazon.in           Standard     Shirt  3XL        Shipped   
2     Amazon     Amazon.in          Expedited     Shirt   XL        Shipped   
3   Merchant     Amazon.in           Standard   Blazzer    L     On the Way   
4     Amazon     Amazon.in          Expedited  Trousers  3XL        Shipped   

   ...  currency  Amount    ship-city   ship-state ship-postal-code  \
0  ...       INR  647.62       MUMBAI  MAHARASHTRA         400081.0   
1  ...       INR  406.00    BENGALURU    KARNATAKA         560085.0   
2  ...       INR  329.00  NAVI MUMBAI  MAHARASHTRA         410210.0   
3  ...       INR  753.33   PUDUCHERRY   PUDUCHERRY         605008.0   
4  ...       INR  574.00      CHENNAI   TAMIL NADU         600073.0   

   ship-country    B2B  fulfilled-by New  PendingS  
0            IN  False     Easy Ship NaN       NaN  
1            IN  False     Easy Ship NaN       NaN  
2            IN   True           NaN NaN       NaN  
3            IN  False     Easy Ship NaN       NaN  
4            IN  False           NaN NaN       NaN  

[5 rows x 21 columns]

### Dataset Shape

In [6]:
df.shape

(128976, 21)

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 128976 entries, 0 to 128975
Data columns (total 21 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   index               128976 non-null  int64  
 1   Order ID            128976 non-null  object 
 2   Date                128976 non-null  object 
 3   Status              128976 non-null  object 
 4   Fulfilment          128976 non-null  object 
 5   Sales Channel       128976 non-null  object 
 6   ship-service-level  128976 non-null  object 
 7   Category            128976 non-null  object 
 8   Size                128976 non-null  object 
 9   Courier Status      128976 non-null  object 
 10  Qty                 128976 non-null  int64  
 11  currency            121176 non-null  object 
 12  Amount              121176 non-null  float64
 13  ship-city           128941 non-null  object 
 14  ship-state          128941 non-null  object 
 15  ship-postal-code    128941 non-nul

### drop unrelated/blank columns

In [8]:

df.drop(['New','PendingS'], axis=1, inplace=True)

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 128976 entries, 0 to 128975
Data columns (total 19 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   index               128976 non-null  int64  
 1   Order ID            128976 non-null  object 
 2   Date                128976 non-null  object 
 3   Status              128976 non-null  object 
 4   Fulfilment          128976 non-null  object 
 5   Sales Channel       128976 non-null  object 
 6   ship-service-level  128976 non-null  object 
 7   Category            128976 non-null  object 
 8   Size                128976 non-null  object 
 9   Courier Status      128976 non-null  object 
 10  Qty                 128976 non-null  int64  
 11  currency            121176 non-null  object 
 12  Amount              121176 non-null  float64
 13  ship-city           128941 non-null  object 
 14  ship-state          128941 non-null  object 
 15  ship-postal-code    128941 non-nul

### Now checking for null or blank values

In [10]:
df.isnull().sum()

index                     0
Order ID                  0
Date                      0
Status                    0
Fulfilment                0
Sales Channel             0
ship-service-level        0
Category                  0
Size                      0
Courier Status            0
Qty                       0
currency               7800
Amount                 7800
ship-city                35
ship-state               35
ship-postal-code         35
ship-country             35
B2B                       0
fulfilled-by          89713
dtype: int64

#### As we can see that currency and amount contain 7800 null values , ship-state , ship-city , ship-postal-code contain 35 null values , fulfilled-by contains 89713 null values.

### We have to remove all those null values for accurate analysis

In [11]:
df.dropna(inplace=True)

In [12]:
df.isnull().sum()

index                 0
Order ID              0
Date                  0
Status                0
Fulfilment            0
Sales Channel         0
ship-service-level    0
Category              0
Size                  0
Courier Status        0
Qty                   0
currency              0
Amount                0
ship-city             0
ship-state            0
ship-postal-code      0
ship-country          0
B2B                   0
fulfilled-by          0
dtype: int64

### As we can see , there is no null value left now .

In [13]:
df['ship-postal-code'].dtype

dtype('float64')

## Now , Changing Data Type for Appropriation

In [14]:
df['ship-postal-code']=df['ship-postal-code'].astype('int')

In [15]:
df['ship-postal-code'].dtype

dtype('int64')

In [16]:
df['Date']=pd.to_datetime (df['Date'])

/tmp/ipykernel_1753/2748755245.py:1: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  df['Date']=pd.to_datetime (df['Date'])


In [17]:
df.columns

Index(['index', 'Order ID', 'Date', 'Status', 'Fulfilment', 'Sales Channel',
       'ship-service-level', 'Category', 'Size', 'Courier Status', 'Qty',
       'currency', 'Amount', 'ship-city', 'ship-state', 'ship-postal-code',
       'ship-country', 'B2B', 'fulfilled-by'],
      dtype='object')

##### Changing column name Qty TO Quantity

In [23]:
df = df.rename(columns={'Qty':'Quantity'})

In [19]:
df.describe()

index                           Date           Qty  \
count   37514.000000                          37514  37514.000000   
mean    60953.809858  2022-05-11 07:56:47.303939840      0.867383   
min         0.000000            2022-03-31 00:00:00      0.000000   
25%     27235.250000            2022-04-20 00:00:00      1.000000   
50%     63470.500000            2022-05-09 00:00:00      1.000000   
75%     91790.750000            2022-06-01 00:00:00      1.000000   
max    128891.000000            2022-06-29 00:00:00      5.000000   
std     36844.853039                            NaN      0.354160   

             Amount  ship-postal-code  
count  37514.000000      37514.000000  
mean     646.553960     463291.552754  
min        0.000000     110001.000000  
25%      458.000000     370465.000000  
50%      629.000000     500019.000000  
75%      771.000000     600042.000000  
max     5495.000000     989898.000000  
std      279.952414     194550.425637

In [20]:
df.describe(include='object')

Order ID                        Status Fulfilment  \
count                 37514                         37514      37514   
unique                34664                            11          1   
top     171-5057375-2831560  Shipped - Delivered to Buyer   Merchant   
freq                     12                         28741      37514   

       Sales Channel ship-service-level Category   Size Courier Status  \
count          37514              37514    37514  37514          37514   
unique             1                  1        8     11              3   
top        Amazon.in           Standard  T-shirt      M        Shipped   
freq           37514              37514    14062   6806          31859   

       currency  ship-city   ship-state ship-country fulfilled-by  
count     37514      37514        37514        37514        37514  
unique        1       4698           58            1            1  
top         INR  BENGALURU  MAHARASHTRA           IN    Easy Ship  
freq      37514       2839         6236        37514        37514

### Describing some specific Columns

In [24]:
df[['Quantity','Amount']].describe()

Quantity        Amount
count  37514.000000  37514.000000
mean       0.867383    646.553960
std        0.354160    279.952414
min        0.000000      0.000000
25%        1.000000    458.000000
50%        1.000000    629.000000
75%        1.000000    771.000000
max        5.000000   5495.000000

## Exploratory Data Analysis (EDA)

In [ ]:
df.columns